src/hidimstat/noise_std.py

```diff
@@ -5,283 +5,256 @@
-            )
-
-        rho_ar = np.zeros(order + 1)
-        rho_ar[0] = 1
-
-        for i in range(1, order + 1):
-            rho_ar[i] = np.median(np.diag(corr_emp, i))
+        residual = np.copy(y)
+        size_support = 0
 
-        A = toeplitz(rho_ar[:-1])
-        coef_ar = solve(A, rho_ar[1:])
+    # estimate the noise standard deviation (eq. 7 in `fan2012variance`)
+    sigma_hat_raw = norm(residual, axis=0) / np.sqrt(n_samples - size_support)
 
-        residual_estimate = np.zeros((n_samples, n_times - order))
-
-        for i in range(order):
-            # time window used to estimate the residual from AR model
-            start = order - i - 1
-            end = -i - 1
-            residual_estimate += coef_ar[i] * residual[:, start:end]
-
-        residual_diff = residual[:, order:] - residual_estimate
-        sigma_eps = np.median(norm(residual_diff, axis=0) / np.sqrt(n_samples))
-
-        rho_ar_full = np.zeros(n_times)
-        rho_ar_full[: rho_ar.size] = rho_ar
-
-        for i in range(order + 1, n_times):
-            start = i - order
-            end = i
-            rho_ar_full[i] = np.dot(coef_ar[::-1], rho_ar_full[start:end])
-
-        corr_hat = toeplitz(rho_ar_full)
-        sigma_hat[:] = sigma_eps / np.sqrt((1 - np.dot(coef_ar, rho_ar[1:])))
-        cov_hat = np.outer(sigma_hat, sigma_hat) * corr_hat
+    if not group:
+        return sigma_hat_raw, beta_hat
 
+    ## Computation of the covariance matrix for group
     else:
-        raise ValueError("Unknown method for estimating the covariance matrix")
-
-    return cov_hat, beta_hat
+        if method == "simple":
+            print("Group reid: simple cov estimation")
+        elif method == "AR":
+            print(f"Group reid: {method}{order} cov estimation")
+            if order > n_times - 1:
+                raise ValueError(
+                    "The requested AR order is to high with "
+                    + "respect to the number of time steps."
+                )
+            elif not stationary:
+                raise ValueError(
+                    "The AR method is not compatible with the non-stationary"
+                    + " noise assumption."
+                )
+        else:
+            raise ValueError("Unknown method for estimating the covariance matrix")
+        ## compute emperical correlation of the residual
+        if stationary:
+            # consideration of stationary noise
+            # (section 2.5 of `chevalier2020statistical`)
+            sigma_hat = np.median(sigma_hat_raw) * np.ones(n_times)
+            # compute rho from the empirical correlation matrix
+            # (section 2.5 of `chevalier2020statistical`)
+            corr_emp = np.corrcoef(residual.T)
+        else:
+            sigma_hat = sigma_hat_raw
+            residual_rescaled = residual / sigma_hat
+            corr_emp = np.corrcoef(residual_rescaled.T)
+
+        # TODO: Why the name of the method is different
```

Comment:
Why the name of the method in group reid is different than the name of the "function"?

Reply:
Not sure I get the question, but the noise estimation in the Desparsified Lasso is by construction, different from that of the Lasso. What we would do would be to study how different the two estimators are and if one outperforms the other.
Likely not in the short term.

Reply:
@bthirion

Reply:
@bthirion

Reply:
I updated the name for the median method from `simple` to `median` for a better meaning of the name.
From your comment, I consider that the name `AR` is enough meaning full.

Reply:
Sounds OK.